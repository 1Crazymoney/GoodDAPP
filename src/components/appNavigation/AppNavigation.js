// @flow
import { createSwitchNavigator } from '@react-navigation/core'
import React from 'react'
import type { Store } from 'undux'

// TODO: Should we do this diferently?
import buySellIcon from '../../assets/buySellIcon.png'
import donateIcon from '../../assets/donateIcon.png'
import homeIcon from '../../assets/homeIcon.png'
import rewardsIcon from '../../assets/rewardsIcon.png'

import GDStore from '../../lib/undux/GDStore'
import Dashboard from '../dashboard/Dashboard'
import Splash from '../splash/Splash'
import BuySell from './BuySell'
import Donate from './Donate'
import Rewards from './Rewards'
import Profile from '../profile/Profile'

import { checkAuthStatus } from '../../lib/login/checkAuthStatus'

type AppNavigationProps = {
  navigation: any,
  store: Store
}

type AppNavigationState = {
  ready: boolean
}

const routes = {
  Rewards: {
    screen: Rewards,
    icon: rewardsIcon
  },
  BuySell: {
    screen: BuySell,
    icon: buySellIcon
  },
  Dashboard: {
    screen: Dashboard,
    icon: homeIcon
  },
  Donate: {
    screen: Donate,
    icon: donateIcon
  },
  Profile: {
    screen: Profile,
    display: false
  }
}

const initialRouteName = 'Dashboard'
const AppNavigator = createSwitchNavigator(routes, { initialRouteName })

/**
 * Switch navigation between all screens on the tabs. Each of this screen should be a StackNavigation
 * Dashboard is the initial route
 */
class AppNavigation extends React.Component<AppNavigationProps, AppNavigationState> {
  checkAuthStatus() {
    if (this.props.store.get('isLoggedInCitizen')) return

    // if not isLoggedInCitizen yet we should check status
    return checkAuthStatus(this.props.store)
  }

  async componentDidMount() {
    await this.checkAuthStatus()
  }

  async componentDidUpdate() {
    await this.checkAuthStatus()
  }

  render() {
    const account = this.props.store.get('account')
    if (account.ready) {
      return <AppNavigator navigation={this.props.navigation} screenProps={{ routes }} />
    }

    return <Splash />
  }
}

const appNavigation = GDStore.withStore(AppNavigation)
appNavigation.router = AppNavigator.router

export default appNavigation
