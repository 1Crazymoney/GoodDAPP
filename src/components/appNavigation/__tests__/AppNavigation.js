import React from 'react'
import AppNavigation from '../AppNavigation'
import { createSwitchNavigator } from '@react-navigation/core'
import { createBrowserApp, Link } from '@react-navigation/web'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('AppNavigation', () => {
  it('renders without errors', () => {
    const Router = createSwitchNavigator({ AppNavigation })
    const WebRouter = createBrowserApp(Router)
    const tree = renderer.create(<WebRouter />)
    expect(tree.toJSON()).toBeTruthy()
  })
})
