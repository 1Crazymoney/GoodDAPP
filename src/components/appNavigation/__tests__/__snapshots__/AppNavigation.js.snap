// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation matches snapshot 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitBoxPack": "center",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "height": "100%",
      "justifyContent": "center",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "center",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msFlexPack": "center",
      "msFlexPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    style={
      Object {
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "height": "80%",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "maxWidth": "100%",
        "minHeight": "256px",
        "minWidth": "212px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "auto",
          "WebkitFlexDirection": "column",
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "backgroundImage": "url(\\"splash.png\\")",
          "backgroundPosition": "center",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "bottom": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "column",
          "flexShrink": 0,
          "height": "100%",
          "left": "0px",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPreferredSize": "auto",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
          "width": "100%",
          "zIndex": -1,
        }
      }
    />
    <img
      alt=""
      draggable={false}
      src="splash.png"
      style={
        Object {
          "bottom": "0px",
          "height": "100%",
          "left": "0px",
          "opacity": 0,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
          "width": "100%",
          "zIndex": -1,
        }
      }
    />
  </div>
</div>
`;
